plugins/generator-1.18.2/forge-1.18.2/templates/block/block.java.ftl

```diff
@@ -459,7 +459,11 @@ public class ${name}Block extends
 	<#if data.requiresCorrectTool>
 	@Override public boolean canHarvestBlock(BlockState state, BlockGetter world, BlockPos pos, Player player) {
 		if(player.getInventory().getSelected().getItem() instanceof TieredItem tieredItem)
-			return tieredItem.getTier().getLevel() >= ${data.breakHarvestLevel};
+			return tieredItem.getTier().getLevel() >= ${data.breakHarvestLevel}
+			<#if data.destroyTool == "pickaxe"> && tieredItem instanceof PickaxeItem

```

Comment:
`PickaxeItem`, `AxeItem` etc. all extend `TieredItem`, so you could replace the class in the first `instanceof` instead of adding another one